### DISCUSSION TITLE
Memory Leak on CreateConnection()

### DISCUSSION THREAD

**Discussion by nbreetz** [rabbitmq/rabbitmq-dotnet-client#1257] [Q&A]
Hey there, hoping someone is able to help.

I have a Xamarin app using RabbitMQ. I've traced a memory leak in our app back to this line:
`_connection = _connectionFactory.CreateConnection();`

When I comment out this line, we don't have the leak. I'm wondering if anyone has experienced this? If so, what did you do to contain the leak? I'm using version 6.4.0.

The leak appears to be fixed when I do a forced garbage collection. This is not best practice however.

Thank you in advance!

**Comment by lukebakken** [comment]
There's not really enough info here to assist much, but here are some questions -
* What are you doing with the connection? I assume you're creating channels, publishing messages or consuming them, etc
* Connections should be created *once* during the lifetime of your application, then disposed when your app is done. How often are you creating connections?

**Reply by nbreetz** [reply]
Thanks for your response Luke.
- We are establishing one connection and one channel. Just consuming messages.
- We are creating one connection when the app launches, and it keeps running while the app is open.

After a few minutes our memory usage slowly starts to increase. This happens with just one connection and before even consuming anything. We're not experiencing the memory issue when the `CreateConnection()` line is commented out. I know this isn't much info to go off of, but anything would help.

**Reply by lukebakken** [reply]
I know it has been a while, but without a way to reproduce this we can't really help. Based on your description you're using the library correctly.
